handle_actions.go

```diff
@@ -12,7 +12,7 @@ import (
 // DoActions do actions based on the context
 func (ms MocksArray) DoActions(c *Context) error {
 	for _, m := range ms {
-		if !c.MatchCondition(m.Expect.Condition) {
+		if !(c.MatchCondition(m.Expect.Condition) && c.MatchConditions(m.Expect.Conditions)) {
```

Comment:
Is there a test that covers the interactions of these two?

Reply:
No, I meant to add one, but there are actually no tests covering `handle_actions.go` -- I will work on adding a few for this